tests/unit/cluster/migrate.tcl

```diff
@@ -0,0 +1,1731 @@
+
+            # Do the snapshot
+            assert_match "OK" [R 2 CLUSTER MIGRATE SLOTSRANGE 16383 16383 NODE $node0_id]
+            set linkname [get_link_name 2 16383]
+            wait_for_migration_field 2 $linkname state waiting-to-pause
+
+            # Multi/Exec should propagate without issue
+            assert_match "OK" [R 2 MULTI]
+            assert_match "QUEUED" [R 2 SET $16383_slot_tag:key1 my_value1]
+            assert_match "QUEUED" [R 2 SET $16383_slot_tag:key2 my_value2]
+            assert_match "QUEUED" [R 2 SET $16383_slot_tag:key3 my_value3]
+            assert_match "OK OK OK" [R 2 EXEC]
+            wait_for_countkeysinslot 0 16383 3
+            wait_for_countkeysinslot 3 16383 3
+
+            # Other databases should SELECT and propagate as expected
+            assert_match "OK" [R 2 SELECT 15]
+            assert_match "OK" [R 2 SET $16383_slot_tag:key1 my_value1]
+            assert_match "OK" [R 2 SELECT 0]
+
+            # Allow migration to complete
+            set_debug_prevent_pause 0
+            wait_for_migration 0 16383
+
+            # Validate MULTI/EXEC
+            assert_match "3" [R 0 CLUSTER COUNTKEYSINSLOT 16383]
+            assert_match "my_value1" [R 0 GET $16383_slot_tag:key1]
+            assert_match "my_value2" [R 0 GET $16383_slot_tag:key2]
+            assert_match "my_value3" [R 0 GET $16383_slot_tag:key3]
+            assert_match "0" [R 2 CLUSTER COUNTKEYSINSLOT 16383]
+
+            # Validate Select
+            assert_match "OK" [R 0 SELECT 15]
+            assert_match "1" [R 0 CLUSTER COUNTKEYSINSLOT 16383]
+            assert_match "my_value1" [R 0 GET $16383_slot_tag:key1]
+            assert_match "OK" [R 0 SELECT 0]
+            assert_match "OK" [R 2 SELECT 15]
+            assert_match "0" [R 2 CLUSTER COUNTKEYSINSLOT 16383]
+            assert_match "OK" [R 2 SELECT 0]
+
+            # Also eventually reflected in replicas
+            wait_for_countkeysinslot 3 16383 3
+            wait_for_countkeysinslot 5 16383 0
+            assert_match "OK" [R 3 SELECT 15]
+            assert_match "1" [R 3 CLUSTER COUNTKEYSINSLOT 16383]
+            assert_match "OK" [R 3 SELECT 0]
+            assert_match "OK" [R 5 SELECT 15]
+            assert_match "0" [R 5 CLUSTER COUNTKEYSINSLOT 16383]
+            assert_match "OK" [R 5 SELECT 0]
+
+            # Migration log shows success on both ends
+            assert {[dict get [get_migration_by_linkname 0 $linkname] state] eq "success"}
+            assert {[dict get [get_migration_by_linkname 2 $linkname] state] eq "success"}
+
+            # Cleanup for the next test
+            assert_match "OK" [R 0 FLUSHALL SYNC]
+            assert_match "OK" [R 0 CLUSTER MIGRATE SLOTSRANGE 16383 16383 NODE $node2_id]
+            wait_for_migration 2 16383
+        }
+    }
+
+    # Test with both migrating slot < existing slot and vice versa, since a lot of the
+    # kvstore logic is ordering dependent
+    foreach testcase [list \
```

Comment:
Can we also add key containment tests on the replica?

Reply:
Oh, I missed that thread. Yeah, basically I was calling out that there was no key-containment.